src/AbstractApplication.php

```diff
@@ -62,6 +67,29 @@ public function __construct(Input $input = null, Registry $config = null)
 		$this->input = $input instanceof Input ? $input : new Input;
 		$this->config = $config instanceof Registry ? $config : new Registry;
 
+		// Create the container
```

Comment:
If we are going to introduce a hard coupling to the DIC then we need to support injection of a Container and the application cannot assume that it is creating these services itself.  Otherwise application bootup scripts like [this one](https://gist.github.com/mbabker/a9b24a472c53f580580e) basically don't work anymore.